group-coordinator/src/main/java/org/apache/kafka/coordinator/group/assignor/RangeSet.java

```diff
@@ -163,6 +171,7 @@ public boolean equals(Object o) {
         if (!(o instanceof Set<?> otherSet)) return false;
 
         if (o instanceof RangeSet other) {
+            if (this.size() == 0 && other.size() == 0) return true;
```

Comment:
If there is no `to < from`, probably we can remove this condition.

Reply:
Sorry, I just got that (5, 5) and (6, 6) are equal in this case. Ignore my comment. Thanks.